lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -207,4 +212,58 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
     current_epoch = Misc.compute_epoch_at_slot(current_slot)
     store.justified_checkpoint.epoch + 1 == current_epoch
   end
+
+  @spec root_by_id(atom() | Types.root() | Types.slot()) ::
+          {:ok, {Types.root(), boolean(), boolean()}} | {:error, String.t()} | atom()
+  def root_by_id(:head) do
```

Comment:
We should make this `root_by_id` function return the block's root, instead of the state root. Then we can easily map that hash to the state root in the calling function. This has the bonus that it's easily reusable, and so it also removes most of the repeated code we can have.

Reply:
do you mean to return `Types.BeaconBlock` ?

Reply:
No. Just `Types.root()`, but the block's hash (you shouldn't compute it, though), instead of the blocks's `state_root`.

Reply:
:thinking: so for example to get  `eth/v1/beacon/states/justified/root` instead of 
```elixir
    with {:ok, justified_checkpoint} <- ForkChoice.get_justified_checkpoint(),
         {:ok, signed_block} <- BlockStore.get_block(justified_checkpoint.root) do
      # TODO compute is_optimistic_or_invalid
      execution_optimistic = true
      {:ok, signed_block.message.state_root, execution_optimistic, false}
    end

```
would be:
```elixir
    with {:ok, justified_checkpoint} <- ForkChoice.get_justified_checkpoint() do
      # TODO compute is_optimistic_or_invalid
      execution_optimistic = true
      {:ok, justified_checkpoint.root, execution_optimistic, false}
    end

```
?

Reply:
Yes. However, after calling `root_by_id` you'd need to call some other function `get_state_root(root)` to get the state root.

Reply:
:thinking: but `get_state_root` is the one that calls `root_by_id` :sweat_smile: 
a similar function that could be used is `StateStore.get_state_root_by_slot(slot)`

Reply:
I created this function that could be useful:
```elixir
  @spec get_state_root(Types.root()) :: {:ok, Types.root()} | {:error, String.t()} | :not_found
  def get_state_root(root) do
    with {:ok, signed_block} <- BlockStore.get_block(root) do
      {:ok, signed_block.message.state_root}
    end
  end
```